### DISCUSSION TITLE
Array Error in Battery RC Data

### DISCUSSION THREAD

**Discussion by c411um77** [OpenModelica/OpenModelica#9713] [Q&A] [answered]
![9F4DE01F-9983-4D91-9CC3-00C03BCF83CB](https://user-images.githubusercontent.com/116268481/202274067-1f9ca239-e76f-48cf-8f9d-406e38cbb999.png)

Hoping someone could help me here :).

When initialising the attached screenshot, I get a failure as my RC parameters in “OneRC” aren’t being read into the CellData parameters. What I get is that the RC data parameter record isn’t an array, but the cell data requires the RC terms to be an array? 

I’ve not used Modelica enough to know what the issue is here. Is anyone able to point me in the right direction? Thanks :)

**Comment by perost** [comment, marked as answer]
`rcData` in the `CellData` class is an array while `OneRC` is a scalar, so `rcData = OneRC` doesn't work. You should use `each rcData = OneRC` or bind it to some array expression such as `rcData = {OneRC}`.
